core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/GroupByAggregator.java

```diff
@@ -131,9 +131,29 @@ public void validate(Schema inputSchema, List<String> groupByFields,
                                            functionInfo.getName(), functionInfo.getField()), null)
           .withConfigElement("aggregates", collectorFieldName);
       }
-    }
 
+      // TODO: CDAP-16401 - Push down validation to individual aggregate functions
+      if (GroupByConfig.Function.COUNTDISTINCT == functionInfo.getFunction()) {
+        if (!conf.containsMacro("aggregates")) {
```

Comment:
is it possible for this to be false at this point? seems like it wouldn't be.

Reply:
ah! the macro check for aggregates is kind of in the weeds so I missed it. Removed it.